src/commands/app/upgrade.tsx

```diff
@@ -0,0 +1,238 @@
+        process,
+        this.apiClient,
+        targetAppVersionCandidates,
+        currentApp,
+        currentAppVersion,
+      );
+    }
+
+    if (!targetAppVersion) {
+      process.error("Target app version could not be determined properly.");
+      return;
+    }
+
+    if (!this.flags.force) {
+      const confirmed: boolean = await process.addConfirmation(
+        <Text>
+          Confirm upgrading {currentApp.name}{" "}
+          {currentAppVersion.externalVersion} (
+          {currentAppInstallation.description}) to version{" "}
+          {targetAppVersion.externalVersion}
+        </Text>,
+      );
+      if (!confirmed) {
+        process.addInfo(<Text>Upgrade will not be triggered.</Text>);
+        process.complete(<></>);
+        ux.exit(1);
+      }
+    } else {
+      process.addInfo(
+        <Text>
+          Commencing upgrade of {currentApp.name}{" "}
+          {currentAppVersion.externalVersion} (
+          {currentAppInstallation.description}) to Version{" "}
+          {targetAppVersion.externalVersion}.
+        </Text>,
+      );
+    }
+
+    const patchAppTriggerResponse =
+      await this.apiClient.app.patchAppinstallation({
+        appInstallationId,
+        data: { appVersionId: targetAppVersion.id },
+      });
+
+    assertStatus(patchAppTriggerResponse, 204);
+
+    let successText: string;
+
+    if (this.flags.wait) {
+      await waitUntilAppStateHasNormalized(
+        this.apiClient,
+        process,
+        appInstallationId,
+        "waiting for app upgrade to be done",
+      );
+      successText =
+        "The upgrade finished successfully. Please check if everything is in its place. 🔎";
+    } else {
+      successText = "The upgrade has been started. Buckle up! 🚀";
+    }
+    await process.complete(<Success>{successText}</Success>);
+  }
+
+  protected render(): ReactNode {
+    return true;
+  }
+}
+
+async function forceTargetVersionSelection(
+  process: ProcessRenderer,
+  apiClient: MittwaldAPIV2Client,
+  targetAppVersionCandidates: AppAppVersion[],
+  currentApp: AppApp,
+  currentAppVersion: AppAppVersion,
+): Promise<AppAppVersion | undefined> {
+  const targetAppVersionString = await process.addSelect(
+    `Please select target upgrade for your ${currentApp.name} ${currentAppVersion.externalVersion} from one of the following`,
+    [
+      ...targetAppVersionCandidates.map((targetAppVersionCandidate) => ({
+        value: targetAppVersionCandidate.externalVersion,
+        label: `${targetAppVersionCandidate.externalVersion}`,
+      })),
+    ],
+  );
+
+  if (targetAppVersionString == "latest") {
+    return await getLatestAvailableTargetAppVersionForAppVersionUpgradeCandidates(
+      apiClient,
+      currentApp.id,
+      currentAppVersion.id,
+    );
+  } else {
+    return await getAvailableTargetAppVersionFromExternalVersion(
+      apiClient,
+      currentApp.id,
+      currentAppVersion.id,
+      targetAppVersionString,
+    );
+  }
```

Comment:
If `latest` is not a selectable option anymore, this condition shouldn't be necessary anymore, either.

```suggestion
  return await getAvailableTargetAppVersionFromExternalVersion(
    apiClient,
    currentApp.id,
    currentAppVersion.id,
    targetAppVersionString,
  );
```

Reply:
latest was removed from the interactive selection only.
You can still trigger an upgrade to `latest` using the  `--target-version` flag (as of now).

Reply:
Should I remove it here aswell? I thought it would be a helpful option especially for custom update scripts.

Reply:
@martin-helmich delete or keep the option to use "latest" for the "target-version"-flag - what's your opinion?
As soon as that is decided eventual changes take about a minute - after that, this feature can be merged.

Reply:
The option to select the latest version via flag is completely valid. But... this function appears to *only* cover interactive selection (the first statement is an unconditional `await process.addSelect`). If `latest` is not an option for interactive selection, I see no way how `targetAppVersionString` could have `"latest"` as value.

Reply:
I was not paying sufficient attention to the line numbers. 
You are completely right and I need to be more careful.

Fixed it immediately.